lib/compress.js

```diff
@@ -323,7 +323,7 @@ merge(Compressor.prototype, {
                     if (side_effects_encountered |= lvalues_encountered) continue;
 
                     // Non-constant single use vars can only be replaced in same scope.
-                    if (ref.scope !== self) {
+                    if (ref.scope.get_defun_scope() !== self) {
```

Comment:
Odd. A top level `figure_out_scope()` should have set the `scope` property on all refs as it does on master branch. I assume this change was necessary to get all `collapse_vars` tests to pass on the harmony branch? If so, something may be amiss with `figure_out_scope()` with new AST classes on that branch.